### PR TITLE
DOCSP-30063: specify return type using withDocument

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-30063
Staging:
- [Specify return type in DB/coll page](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-30063-withDocument-method/fundamentals/databases-collections/#specify-return-type)
- [new content in data class doc format pg](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-30063-withDocument-method/fundamentals/data-formats/document-data-format-data-class/#retrieve-a-data-class)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: examples/src/test/kotlin/DatabaseCollectionsTest.kt [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 import kotlinx.coroutines.flow.toList
 import kotlinx.coroutines.runBlocking
 import org.bson.codecs.pojo.annotations.BsonId
 import org.bson.types.ObjectId
 import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
+import java.util.concurrent.TimeUnit
+import kotlin.test.Ignore
```
Comment (cbullinger) [inline examples/src/test/kotlin/DatabaseCollectionsTest.kt:14]: is this actually being used?

File: source/fundamentals/databases-collections.txt [hunk 2]
```diff
@@ -96,15 +98,55 @@ against a series of filters during writes to a collection. You can
...
+
+- Your collection contains multiple data types.
+- You specify a projection that changes your data fields.
+- You cannot directly specify a return type on a method that changes the data,
+  such as ``findOneAndUpdate()`` or ``findOneAndReplace()``.
+
+The following example shows how you can retrieve a collection that
+contains represented by the ``Fruit`` data class but return the result
+of a ``findOneAndUpdate()`` operation as an instance of the ``NewFruit``
+class:
```
Comment (cbullinger) [inline source/fundamentals/databases-collections.txt:133]: suggestion: mention the renamed field? it honestly took me a second to realize the difference between `Fruit` and `NewFruit`

**Review by cbullinger — APPROVED** [review]
This looks great! Nice job!

File: source/fundamentals/databases-collections.txt [hunk 3]
```diff
@@ -14,23 +14,25 @@ Overview
...
 
 #. **Collections**: Databases are organized into collections which contain **documents**.
 
-#. **Documents**: Documents contain literal data such as strings, numbers, and dates, as well as other embedded documents. For more information on document field types and structure, see the :manual:`server documentation for documents </core/document/>`. 
+#. **Documents**: Documents contain literal data such as strings, numbers, and dates, as well as other embedded documents. For more information on document field types and structure, see the :manual:`Server documentation on documents </core/document/>`.
 
-With the MongoDB Kotlin driver, data can be modeled using Kotlin data classes or you can use a 
-`Document <{+api+}/apidocs/bson/org/bson/Document.html>`__ class
-to store and retrieve data from MongoDB.
+With the MongoDB Kotlin driver, You can model data by using Kotlin data
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/databases-collections.txt:27]: facepalm

File: source/fundamentals/databases-collections.txt [hunk 4]
```diff
@@ -96,15 +98,55 @@ against a series of filters during writes to a collection. You can
...
+Specifying a different return class could be useful in the following
+situations:
+
+- Your collection contains multiple data types.
+- You specify a projection that changes your data fields.
+- You cannot directly specify a return type on a method that changes the data,
+  such as ``findOneAndUpdate()`` or ``findOneAndReplace()``.
+
+The following example shows how you can retrieve a collection that
+contains represented by the ``Fruit`` data class but return the result
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/databases-collections.txt:131]: facepalm again

File: examples/src/test/kotlin/DatabaseCollectionsTest.kt (hunk 1 above)
Comment (rustagir) [reply examples/src/test/kotlin/DatabaseCollectionsTest.kt:14]: WIll remove any unused imports from both code files relating to this PR

File: source/fundamentals/databases-collections.txt (hunk 2 above)
Comment (rustagir) [reply source/fundamentals/databases-collections.txt:133]: added descriptions of the operation to both pages.

**Comment by rustagir** [comment]
I moved the subsection on the DB/collection page from the create coll section to the access a coll section, as the method applies when you are accessing a collection, not when you are creating one.

**Review by rozza — APPROVED** [review]
LGTM

**Pull request merged**